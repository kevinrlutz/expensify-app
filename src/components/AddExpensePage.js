import React from 'react'
import ExpenseForm from './ExpenseForm'
import { connect } from 'react-redux'
import { addExpense } from '../actions/expenses'

export class AddExpensePage extends React.Component {
	onSubmit = (expense) => {
		this.props.onSubmit(expense)
		this.props.history.push('/')
	}
	render() {
		return (
			<div>
				<div className="page-header">
					<div className="content-container">
						<h1 className="page-header__title">Add Expense</h1>
					</div>
				</div>

				<div className="content-container">
					<ExpenseForm
						onSubmit={(expense) => {
							this.props.addExpense(expense)
							this.props.history.push('/')
						}}
					/>
				</div>
			</div>
		)
	}
}

const mapDispatchToProps = (dispatch) => ({
	addExpense: (expense) => dispatch(addExpense(expense)),
})

export default connect(undefined, mapDispatchToProps)(AddExpensePage)
