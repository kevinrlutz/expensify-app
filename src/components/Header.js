import React from 'react'
import { Link as NavLink } from 'react-router-dom'

const Header = () => {
	return (
		<header className="header">
			<div className="content-container">
				<div className="header__content">
					<NavLink to="/" className="header__title">
						<h1>Expensify</h1>
					</NavLink>
					<button>Log Out</button>
				</div>
			</div>
		</header>
	)
}

export default Header
