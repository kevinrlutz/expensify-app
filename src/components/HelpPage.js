import React from 'react'

function HelpPage() {
	return <main>HelpPage</main>
}

export default HelpPage
