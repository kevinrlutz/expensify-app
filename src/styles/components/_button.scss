.button {
	background: $blue;
	color: white;
	border: none;
	display: inline-block;
	text-decoration: none;
	line-height: 1;
	padding: $s-size;
	font-weight: 300;
	font-size: $font-size-l;
}

.button--secondary {
	@extend .button;
	background: #888888;
}
