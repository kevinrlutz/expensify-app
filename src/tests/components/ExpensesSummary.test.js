import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'

test('should render ExpensesSummary for no expenses correctly', () => {
	const wrapper = shallow(
		<ExpensesSummary expenseCount={1} expensesTotal={235} />
	)
	expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary for one expense correctly', () => {
	const wrapper = shallow(
		<ExpensesSummary expenseCount={23} expensesTotal={1520002} />
	)
	expect(wrapper).toMatchSnapshot()
})
