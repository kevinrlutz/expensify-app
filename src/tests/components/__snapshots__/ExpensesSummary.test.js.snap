// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary for no expenses correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totaling
       
      <span>
        $2.35
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary for one expense correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
       totaling
       
      <span>
        $15,200.02
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
